### PR TITLE
Feature: Carnival QoL

### PR DESCRIPTION
**Pull request by Thunderblade73**
## What
Some needed QoL for Carnival. See Changelog for details.

<details>
<summary>Images</summary>

</details>

## Changelog New Features
+ Added a Reminder to claim Carnival Tickets. - Thunderbalde73
+ Added display for Carnival Goals. - Thunderbalde73
+ Added a feature to double-click the Carnival NPC to start the game without clicking in the chat. - Thunderbalde73

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/event/carnival/CarnivalReminder.kt [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    fun check() {
+        if (claimedToday) {
+            val currentDay = ZonedDateTime.now(ZoneOffset.UTC).toLocalDate()
+            val lastClaimedDay = lastClaimedDay
+
+            if (lastClaimedDay == null || currentDay.isAfter(lastClaimedDay)) {
+                claimedToday = false
+            }
+        } else {
+            ChatUtils.clickableChat(
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/event/carnival/CarnivalReminder.kt]: Please use `ChatUtils.clickToActionOrDisable()` instead here.

Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/event/carnival/CarnivalReminder.kt]: Also maybe even check against `ReminderUtils.isBusy()`.

**Review by hannibal002 — CHANGES_REQUESTED** [review]
Double click NPC works fine.

Have not noticed a daily ticket reminder message.

This looks wrong to me:
image

**Comment by Thunderblade73** [comment]
> Double click NPC works fine.
> 
> Have not noticed a daily ticket reminder message.
> 
> This looks wrong to me: image

There is the issue
image
(Wrong copy paste)

File: src/main/java/at/hannibal2/skyhanni/features/event/carnival/CarnivalReminder.kt (hunk 1 above)
Comment (Thunderblade73) [reply src/main/java/at/hannibal2/skyhanni/features/event/carnival/CarnivalReminder.kt]: Changed. And worked. (5min is just quiet long)
image

**Review by hannibal002 — CHANGES_REQUESTED** [review]
image

**Review by hannibal002 — CHANGES_REQUESTED** [review]
image

**Review by hannibal002 — CHANGES_REQUESTED** [review]
<details>
<summary>Fruit digging: Dragonfruit</summary>

image
```
§f    §r§f4x §r§aApple§r§f, §r§f3x §r§aMango§r§f, §r§f3x §r§9Watermelon§r§f, §r§f1x §r§5Cherry§r§f, §r§f2x §r§5Coconut§r§f,
§r§f                    §r§f1x §r§dDragonfruit§r§f, §r§f2x §r§cBomb§r§f, §r§f2x §r§eRum
```

</details>

**Comment by Thunderblade73** [comment]
> image

You already have it done. That's why there is no message for  the complete of the goal from hypixel. This is intended behavior. Talk with Camilla to update your already completed goals. (Same for all other 2 issues)

**Review by hannibal002 — APPROVED** [review]
Talking to Chantelle updates the goals, yes

**Pull request merged**